### PR TITLE
[2.9.0] 1373 support authentication with service account tokens

### LINKED ISSUE #1373: Support authentication with service account tokens
## Related Issues

(https://github.com/rancher/rancher/issues/22417)
https://jira.suse.com/browse/SURE-2476

## Summary
Ranchers auth proxy can now support authentication of requests that specify a Service Account token in the Authorization Bearer header

### PR DESCRIPTION
**Pull request by LucasSaintarbor**
Fixes #1373 

## Reminders

- See the [README](../README.md) for more details on how to work with the Rancher docs.

- Verify if changes pertain to other versions of Rancher. If they do, finalize the edits on one version of the page, then apply the edits to the other versions.

- If the pull request is dependent on an upcoming release, remember to add a "MERGE ON RELEASE" label and set the proper milestone.

## Description

From issue:

Ranchers auth proxy can now support authentication of requests that specify a Service Account token in the Authorization Bearer header. 

More info/context: 

JWT Authentication is also known as Service Account Token Authentication.

This feature, when enabled, lets a user set up a downstream cluster to support authentication, through Rancher, of tokens that are created for a service account that exists on a downstream cluster (those tokens are in the form of a JWT).

Prior to this feature, Rancher would reject such requests because Rancher would only support Rancher-issued tokens (which are NOT JTWs). Some users worked-around this limitation by issuing those requests directly to the downstream cluster, rather than relying on Rancher's auth/security. With this feature enabled, users no longer have to work-around Rancher.

A common use case for this is to enable integration of secret vault solutions (like Hashicorp Vault). You can see the original rancher/rancher issue for more details https://github.com/rancher/rancher/issues/22417.

## Comments / Questions

- Referencing https://github.com/rancher/rancher-docs/issues/1373#issuecomment-2243357194, @crobby may you provide the steps to manually create a ClusterProxyConfig object in the target clusters namespace on the local cluster to enable/disable the feature?

### CODE REVIEW

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/jwt-authentication.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+title: JWT Authenticaiton
```
Comment (martyav) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/jwt-authentication.md:2]: Probably should expand the acronym in the title. In-text references expand the acronym on first use, then use JWT throughout

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/jwt-authentication.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+---
+<!-- Edit Canonical Link -->
+<head> 
+  <link rel="canonical" href="https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/jwt-authentication"/>
+</head>
+
+Many 3rd party integrations available for Kubernetes, such as Gitlab and Hashicorp Vault, involve giving an external process access to the Kubernetes API using a native Kubernetes Service Account token for authentication.
+
+In Rancher v2.9.0 and later, you can enable a downstream cluster to support JWT authentication of tokens created for a service account on a downstream cluster through the Rancher authentication proxy. Prior to this feature, Rancher rejects such requests because Rancher only supports Rancher-issued tokens.
```
Comment (martyav) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/jwt-authentication.md:11]:
```suggestion
In Rancher v2.9.0 and later, you can enable a downstream cluster to support JSON web token (JWT) authentication of tokens created for a service account on a downstream cluster through the Rancher authentication proxy. In Rancher versions earlier than  v2.9.0, only Rancher-issued tokens were supported.
```

I'm having some trouble making "you can enable a downstream cluster to support JSON web token (JWT) authentication of tokens created for a service account on a downstream cluster through the Rancher authentication proxy" shorter and easier to read. I feel like this should probably be split into two sentences, or reorganized so there aren't as many clauses. Something like, "service accounts on downstream clusters can now authenticate through JWT using the Rancher authentication proxy", if I'm understanding the sentence correctly.

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/jwt-authentication.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+:::note
+JWT authentication is also known as service account token authentication.
+:::
```
Comment (martyav) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/jwt-authentication.md:15]:
```suggestion
JWT authentication is also known as service account token authentication.
```

Is that generally true, or is it just that Rancher uses JWT for service account authentication?

**Review by martyav — CHANGES_REQUESTED** [review]
I left some comments alongside edit suggestions.

**Review by btat — CHANGES_REQUESTED** [review]

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/jwt-authentication.md (hunk 3 above)
Comment (crobby) [reply docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/jwt-authentication.md:15]: Not strictly true.
Service Account Token Auth happens to be JWT-based (and in the original github issue, we call this feature Service account token auth), but to say that JWT auth is a.k.a. Service account token auth would be incorrect.  JWTs have many other uses outside of Rancher or service account token auth.

Comment (LucasSaintarbor) [reply docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/jwt-authentication.md]: @crobby Thanks for confirming. I removed this.

**Comment by LucasSaintarbor** [comment]
@crobby May you provide a review when you get the chance? Thanks!

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/jwt-authentication.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+---
+title: JSON Web Token (JWT) Authentication
+---
+<!-- Edit Canonical Link -->
```
Comment (btat) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/jwt-authentication.md:4]:
```suggestion

```

File: versioned_docs/version-2.9/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/jwt-authentication.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
+---
+title: JSON Web Token (JWT) Authentication
+---
+<!-- Edit Canonical Link -->
```
Comment (btat) [inline versioned_docs/version-2.9/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/jwt-authentication.md:4]:
```suggestion

```

**Review by btat — CHANGES_REQUESTED** [review]

**Review by btat — APPROVED** [review]

**Review by crobby — APPROVED** [review]
I think this is good.  Accurate and concise.

**Review by martyav — APPROVED** [review]

**Pull request merged**